pyproject.toml

```diff
@@ -171,8 +168,8 @@ datadog = "^0.43.0"
 [tool.poetry.extras]
 spacy = [ "spacy",]
 jieba = [ "jieba",]
-transformers = [ "transformers",]
-full = [ "spacy", "transformers", "jieba",]
+transformers = [ "transformers", "sentencepiece"]
```

Comment:
So `sentencepiece` is an optional dependency on `transformers` side and hence we also need to add it as a dependency?

Reply:
yep - I can double check one last time, but I think we had trouble when we didn't include it explicitly this way

Reply:
@usc-m could you double check this was a required dep by transformer that won't be installed otherwise? For me it looks fine here, can't get it to complain

Reply:
It's a required dep for the models we use, for the previous version of `transformers` it was a required dependency, but for the newer one it's an extra, so we can install it with `transformers[sentencepiece]` (which I just discovered confirming this, and would respect their constraints on the version of `sentencepiece` better) or add it as an explicit dependency like this. (They do their dependencies in a really strange way so it's very non-obvious this is the case)

https://github.com/huggingface/transformers/blob/master/setup.py#L269 (current)
vs
https://github.com/huggingface/transformers/blob/v2.11.0/setup.py#L125 (previous)

Reply:
As a side note that's maybe more useful for research to know, this means that `transformers` is now (in new versions) installed in a 'lightweight' way where the dependencies for some models are not installed by default. So you may encounter runtime failures due to missing dependencies because of this

Reply:
We should put this comment in the code, I think. Maybe above the list of the pretrained models? - Or a notion page on transformer related things?

Reply:
> installed in a 'lightweight' way where the dependencies for some models are not installed by default. So you may encounter runtime failures due to missing dependencies because of this

Would it work for at least the pre-trained models that we support officially and mention in the[ docs](https://rasa.com/docs/rasa/components/#languagemodelfeaturizer)?

Reply:
@dakshvar22 I don't think so, [looks like](https://huggingface.co/docs/transformers/tokenizer_summary#sentencepiece) we need it for XLNet?